File: C++/1898. 【认证试题】遥控小车.cpp
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2020. All rights reserved.
 * Description: 上机编程认证
 * Note: 缺省代码仅供参考，可自行决定使用、修改或删除
 */
#include <iostream>
#include <string>
#include <utility>
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    // 待实现函数，在此函数中填入答题代码;
    string ExecCommand(const string &commands)
    {
        string result;
        vector<int> res = {0, 0};
        int flag = commands.size() - commands.size() % 4 + 4;
        
        for(auto c:commands) {
            switch(c) {
                case 'G':
                    res[flag%2] += (flag % 4 / 2 == 0? 1 : -1);
                    break;
                case 'L':
                    flag -= 1;
                    break;
                case 'R':
                    flag += 1;
                    break;
            }
        }
        
        result = '(' + to_string(res[1]) + ',' + to_string(res[0]) + ')';
        
        return result;
    }
    
};

inline string ReadLine()
{
    string line;
    getline(cin, line);
    return line;
}

int main()
{   
    string commands = ReadLine();
    
    Solution solu;
    string result = solu.ExecCommand(commands);
    cout << result;
    return 0;
}

File: Python/1898. 【认证试题】遥控小车.py
"""
Copyright (c) Huawei Technologies Co., Ltd. 2020-2020. All rights reserved.
Description: 上机编程认证
Note: 缺省代码仅供参考，可自行决定使用、修改或删除
"""


class Solution:
    def exec_command(self, commands):
        res = [0, 0]
        flag = len(commands) - len(commands) % 4 + 4
        for c in commands:
            if c == 'L':
                flag -= 1
            elif c == 'R':
                flag += 1
            else:
                res[flag % 2] += 1 if flag % 4 // 2 == 0 else -1
        return reversed(res)


if __name__ == "__main__":
    commands = str(input().strip())
    function = Solution()
    results = function.exec_command(commands)
    print("(" + str.join(",", map(str, results)) + ")")

